src/sinks/aws_kinesis/firehose/integration_tests.rs

```diff
@@ -60,6 +60,7 @@ async fn firehose_put_records() {
         auth: Default::default(),
         acknowledgements: Default::default(),
         request_retry_partial: Default::default(),
+        partition_key_field: None,
```

Comment:
Currently we don't have int tests covering partition key for firehose, because it wasnt used previously. Now that it is though, we should add integration tests for it.

Note this should be pretty straightforward based off of `kinesis_put_records_with[out]_partition_key` int tests for streams.

I have a PR open fixing the previously disabled integration test for streams to use as a reference: https://github.com/vectordotdev/vector/pull/19220

Reply:
thanks @neuronull for your reply, I added this test as you request